compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -344,20 +344,19 @@ object JavaParsers {
       */
     def annotation(): Option[Tree] = {
       val id = convertToTypeId(qualId())
-      var skipAnnot = false
+      var annot: Option[Tree] = Some(ensureApplied(Select(New(id), nme.CONSTRUCTOR)))
```

Comment:
It's better to start with None and only construct the value when we're sure we're going to use it.

Reply:
Is this ok?
```scala
      var annot: Option[Tree] = None

      // only parse annotations without arguments
      if (in.token == LPAREN && in.lookaheadToken != RPAREN) {
        skipAhead()
        accept(RPAREN)
      }
      else {
        if (in.token == LPAREN) {
          in.nextToken()
          accept(RPAREN)
        }
        annot = Some(ensureApplied(Select(New(id), nme.CONSTRUCTOR)))
      }

      annot
```

Reply:
in fact we don't even need the var:
```scala
      // only parse annotations without arguments
      if (in.token == LPAREN && in.lookaheadToken != RPAREN) {
        skipAhead()
        accept(RPAREN)
        None
      }
      else {
        if (in.token == LPAREN) {
          in.nextToken()
          accept(RPAREN)
        }
        Some(ensureApplied(Select(New(id), nme.CONSTRUCTOR)))
      }
```

Reply:
Thanks! I pushed the changes.